docs/DeveloperGuide.md

```diff
@@ -1,34 +1,890 @@
+
+##### Implementation
+When the user attempts to list the details of a particular recipe, the `listRecipeCommand`, ‘Parser’ and `Recipe` class will be accessed and the following sequence of actions are called to list details of  a particular `recipe` object:
+1. User executes `listrecipe 1`  
+    2. A `Ui` object will be created and calls `Ui#getUserCommand()`
+    3. Input will be parsed in `Command#parseUserCommand()` and identified with the keyword `listrecipe`.
+    ![List Ingredient Step 1](images/AddRecipe1.png)
+2. Parsing of user input and creation of command object
+    2.This will automatically trigger the parsing of the user’s input string into a suitable format for the listing of `recipe` object in `Command#prepareListRecipe()`.
+    3. A `ListRecipeCommand` object will be created.
+    ![List Ingredient Step 2](images/ListRecipeCommand.png)   
+3. Executing Command
+    2. The newly created object will call `ListRecipeCommand#execute` which starts the process of listing a particular recipe’s details, thus, calling `ListRecipeCommand#listRecipe()`.
+    3. The existing recipeList arraylist and the item number of the chosen recipe will be passed through to the `ListRecipeCommand#listRecipe()`.
+    4. The function will find if the item number is valid and contains details of the recipe, thus, creates a CommandResult storing the details of the particular recipe.
+    ![List Ingredient Step 3](images/ListRecipeCommand2.png)
+4. The details will then be printed onto the console using `Ui#showResultToUser(result)`.
+    
+
+##### Design Considerations
+Aspect: Finding the recipe requested by the user.
+
+Alternative 1: Looping through the whole recipeList arraylist to find the recipe requested by the user.
+
+|     |     |
+|-----|-----|
+|**Pros** | The program will be able to locate the recipe accurately. |
+|**Cons** | This method will be slow when facing a huge amount of data in the arraylist as the program may have to go through every single item in the arraylist. |
+
+Alternative 2 (current choice): Using arrayList.get(item) to get the recipe requested by the user.
+
+|     |     |
+|-----|-----|
+|**Pros** | Users would be able to get the details of the particular recipe accurately and fast. |
+|**Cons** | Without proper checks done before running the command, it will result in error if the number indicated by the user exceeds the arraylist / does not exist in the arraylist.  |
+
+#### 4.2.3. Cooking of recipe
+The feature allows the user to cook a recipe if there are sufficient ingredients. The user will also indicate how many pax this recipe would be cooked for.
+
+![Cook Recipe Sequence Diagram](images/cookRecipeCommandSequenceDiagram.png)
```

Comment:
Is the setQuantityCommand() to the Ingredient object supposed to be inside the opt frame?

Reply:
Seems like right now, the opt frame has nothing in it